core/tx_list.go

```diff
@@ -506,8 +506,12 @@ func (l *txList) Filter(costLimit *uint256.Int, gasLimit uint64) (types.Transact
 				lowest = nonce
 			}
 		}
+		l.txs.m.Lock()
```

Comment:
Could you use Trylock? in case that the lock already happened.

Reply:
Line 492 calls the Filter() method which locks (and defers unlock) the lock in question so it should be safe to assume the lock isn't already held by this go routine (if it were, it'd deadlock on 492).  I'm not particularly familiar w/ TryLock though - should I still add it?